src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/attachscanneddocs/AttachScannedDocsAboutToSubmitHandler.java

```diff
@@ -40,22 +41,29 @@ public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType,
         if (deletedRedactedDocEnabled) {
             var scannedDocuments = callback.getCaseDetails().getCaseData().getScannedDocuments();
 
-            callback.getCaseDetailsBefore().ifPresent(
-                    sscsCaseDataCaseDetailsBefore -> {
-                        sscsCaseDataCaseDetailsBefore.getCaseData().getScannedDocuments()
-                                .stream()
-                                .filter(scannedDocumentBefore -> scannedDocumentBefore.getValue().getEditedUrl() != null)
-                                .forEach(scannedDocumentBefore -> {
-                                    scannedDocuments
-                                            .stream()
-                                            .forEach(scannedDocument -> {
-                                                if (compare(scannedDocumentBefore, scannedDocument)) {
-                                                    scannedDocument.getValue().setEditedUrl(scannedDocumentBefore.getValue().getEditedUrl());
-                                                }
-                                            });
-                                });
+            if (scannedDocuments != null) {
+                callback.getCaseDetailsBefore().ifPresent(
+                        sscsCaseDataCaseDetailsBefore -> {
+                            var scannedDocumentsBefore = sscsCaseDataCaseDetailsBefore.getCaseData().getScannedDocuments();
+
+                            if (scannedDocumentsBefore != null) {
+                                scannedDocumentsBefore.stream()
+                                        .filter(scannedDocumentBefore -> scannedDocumentBefore.getValue().getEditedUrl() != null)
+                                        .forEach(scannedDocumentBefore -> {
+                                            scannedDocuments
+                                                    .stream()
+                                                    .forEach(scannedDocument -> {
+                                                        if (compare(scannedDocumentBefore, scannedDocument)) {
+                                                            scannedDocument.getValue().setEditedUrl(scannedDocumentBefore.getValue().getEditedUrl());
+                                                        }
+                                                    });
+                                        });
 
-                    });
+                            }
+                        });
+            } else {
+                log.error("NullPointerException occurred in handle method");
```

Comment:
we should just say that there are no scanned documents in this case (we stopped the NPE from occurring)